scale/job/configuration/interface/job_interface_1_0.py

```diff
@@ -443,6 +443,17 @@ def _check_param_name_uniqueness(self):
                 raise InvalidInterfaceDefinition('shared resource & input_data names must be unique')
             self._param_names.add(shared_resource['name'])
 
+        if self.definition['version'] >= '1.2':
+            for setting in self.definition['settings']:
+                if setting['name'] in self._param_names:
+                    raise InvalidInterfaceDefinition('Setting names must be unique')
+                self._param_names.add(setting['name'])
+
+            for env_var in self.definition['env_vars']:
+                if env_var['name'] in self._param_names:
+                    raise InvalidInterfaceDefinition('Enviorment variable names must be unique')
+                self._param_names.add(env_var['name'])
+
```

Comment:
This code should be in the 1.2 schema, not 1.0.

While settings do have to be unique against all of the other params, the env var names only have to be unique compared to each other.